src/Sarif/Writers/BaseLogger.cs

```diff
@@ -64,6 +75,33 @@ public bool ShouldLog(Notification notification)
 
         public bool ShouldLog(Result result)
         {
+            if (_resultsCount != null)
+            {
+                Location firstLocation = result.Locations.First();
+                if (firstLocation.PhysicalLocation == null ||
+                    result.Run?.OriginalUriBaseIds == null ||
+                    !firstLocation.PhysicalLocation.ArtifactLocation.TryReconstructAbsoluteUri(result.Run.OriginalUriBaseIds, out Uri locationUri))
+                {
+                    locationUri = firstLocation.PhysicalLocation.ArtifactLocation.Uri;
+                }
+                var resultKey = new Tuple<string, Uri>(result.RuleId, locationUri);
+
+                if (_resultsCount.TryGetValue(resultKey, out int currentCount))
+                {
+                    if (currentCount >= _resultsLimitPerRuleTarget)
+                    {
+                        ++_resultsLimited;
```

Comment:
>++_resultsLimited; [](http://example.com/codeflow?start=24&length=18)

Should we add a notification saying: "ruleX fileY reached limit"? This would allow us to understand why we have so many findings of that type.

Reply:
I like the idea of notification, but will adding a notification here trigger one each time the results are limited? That would cause a lot of noisy notifications. I think when the results are finalized, we can send just one notification for the rule that says the results for this rule have been truncated

Reply:
I added one notification per rule per file.